source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveActionSetRenderer.cs

```diff
@@ -89,8 +89,15 @@ public static void AddRenderedActions(Grid uiContainer, IList<AdaptiveAction> ac
 
                 foreach (AdaptiveAction action in actionsToProcess)
                 {
+                    var rendereableAction = context.GetRendereableElement(action);
+
+                    if (rendereableAction is AdaptiveSubmitAction)
+                    {
+                        context.SubmitActionCardId[rendereableAction as AdaptiveSubmitAction] = context.RenderArgs.ContainerCardId;
```

Comment:
nit: is there a reason why SubmitActionCardId has AdaptiveSubmitAction as a key?

Reply:
I had to use this as it's the only data we have when the action is invoked

---
In reply to: [440532423](https://github.com/microsoft/AdaptiveCards/pull/4172#discussion_r440532423) [](ancestors = 440532423)